### DISCUSSION TITLE
Auto scrape meta data?

### DISCUSSION THREAD

**Discussion by argash** [advplyr/audiobookshelf#291] [Ideas]
First off I FREAKING LOVE THIS PROJECT!!! Seriously I can't stand Calibre and this is already 1000x more useful for me! So the only thing I'm wondering is if we could get a feature to specify our preferred meta data scraper and book cover scraper? I ask because generally I always want the data from Audible and the cover from Google or Open Library. I may still manually adjust a few files here and there. In general though if it would auto scrape for me I wouldn't need to adjust very much.

**Comment by advplyr** [comment]
Yeah this is going to happen. First I wanted to make sure the Match feature was working well.
The next step is to add some server settings to enable auto-populate metadata with your preferred sources.
The difficult part is how the metadata is prioritized. I added 2 temporary server settings to prefer ID3 tag metadata or prefer OPF file metadata, but ideally this is all managed in a nice interface.

If you have any suggestions on specific implementations or how this looks like on the config page please share

**Comment by argash** [comment]
A basic implementation would let me choose preferred scrapper for meta data and covers individually, as well as prefering square or standard size covers. 

I was also thinking maybe their should be two fields for publication date? One for print and one for audio? I know it really only affects older titles. But even newer stuff like the Martian has had two separate audio releases (One with RC Bray and one with Wil Wheaton).

**Reply by advplyr** [reply]
It may need to be more specific about which metadata fields are used. For example my library works best with Audible metadata, but I don't want to include the narrator.
Audible metadata also includes a series, but I don't think this should override the series folder you put your audiobook in.

**Reply by argash** [reply]
Yea, I think the "settings" for auto scrapping could be as detailed or advanced as you want to make them. In fact may want to have a basic set of settings that's just like "default agent" then a toggle for advance settings where you could set a priority order for agents for each field.

**Reply by advplyr** [reply]
Just released `v1.6.55` that allows you to choose the provider you want to use for finding covers.

**Reply by argash** [reply]
Seems to be working

**Comment by advplyr** [comment]
I forgot to mention on this discussion that support for quick matching metadata on all books in a library was added. Details here: #359
